pkg/controller/statefulset/stateful_pod_control.go

```diff
@@ -241,9 +241,10 @@ func (spc *StatefulPodControl) UpdatePodClaimForRetentionPolicy(set *apps.Statef
 			return fmt.Errorf("Could not retrieve claim %s not found for %s when checking PVC deletion policy: %w", claimName, pod.Name, err)
 		default:
 			if !claimOwnerMatchesSetAndPod(claim, set, pod) {
-				needsUpdate := updateClaimOwnerRefForSetAndPod(claim, set, pod)
+				claimClone := claim.DeepCopy()
```

Comment:
I think a deep copy is already made in UpdateStatefulSet: https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/statefulset/stateful_set_control.go#L78

So the problem just occurs during tests. Can a deep copy be made in the test, so we don't have duplicated copies in production?

Reply:
I'm afraid not. This `DeepCopy` I added is for the pvc instead of statefulset, and this pvc is gotten form informer directly,  https://github.com/kubernetes/kubernetes/blob/a18c1f71d25d5fc9905d17054bf4a4a4523e8b76/pkg/controller/statefulset/stateful_pod_control.go#L236

So if we ignore this `DeepCopy`, the informer cache about the pvc will be altered.

Reply:
Oh! good point, thanks.

/lgtm

Reply:
(we need someone else to approve)

Reply:
@soltysh  @smarterclayton  we need approve, thanks!